### PR TITLE
MODORDERS-350 Update piece schema (receipt date)

### PR DESCRIPTION
**Pull request by pautau-s**
## Purpose
[MODORDERS-350](https://issues.folio.org/browse/MODORDERS-350)

## Approach
- verify existence of piece.receipDate upon POST piece

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/rest/impl/InventoryHelper.java [hunk 1]
```diff
@@ -303,6 +303,7 @@ private String buildLookupEndpoint(String type, Object... params) {
                 .map(itemId -> new Piece().withFormat(pieceFormat)
                                           .withItemId(itemId)
                                           .withPoLineId(polId)
+                                          .withReceiptDate(compPOL.getReceiptDate())
```
Comment (KVupp) [inline src/main/java/org/folio/rest/impl/InventoryHelper.java]: the receipt data on compPOL set after the line is entirely received, it'll always be null here
https://github.com/folio-org/mod-orders/blob/e1370f278e30ad4d4d52be7c7b8f9e785d6adae9/src/main/java/org/folio/orders/utils/HelperUtils.java#L1117

Comment (pautau-s) [reply src/main/java/org/folio/rest/impl/InventoryHelper.java]: Hmm. Do you know where the value comes from?

Comment (KVupp) [reply src/main/java/org/folio/rest/impl/InventoryHelper.java]: As I understand it has to come from the UI while checkin 
Screen Shot 2020-01-09 at 4 34 56 PM

Comment (KVupp) [reply src/main/java/org/folio/rest/impl/InventoryHelper.java]: receipt_date is a mandatory field only while creating pieces using checkin, as pieces for receiving don't come from the UI, 
1) either populate it with some arbitrary date 
2) that date cannot be mandatory on schema, we just have a check on BL

**Pull request merged**